OutOfSchool/OutOfSchool.DataAccess/Repository/Files/GcpFilesStorageBase.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.Storage.V1;
+using OutOfSchool.Services.Common.Exceptions;
+using OutOfSchool.Services.Contexts.Configuration;
+using OutOfSchool.Services.Extensions;
+using OutOfSchool.Services.Models;
+
+namespace OutOfSchool.Services.Repository.Files
+{
+    public abstract class GcpFilesStorageBase<TFile> : IFilesStorage<TFile, string>
+        where TFile : FileModel, new()
+    {
+        private readonly StorageClient storageClient;
+        private readonly string bucketName;
+
+        protected GcpFilesStorageBase(GcpStorageSourceConfig options)
+        {
+            var credential = options.RetrieveGoogleCredential();
+            storageClient = StorageClient.Create(credential);
```

Comment:
This looks like hidden dependency not covered by DI. Not sure if it should be or not, but current implementation makes testing of GcpFilesStorageBase descendants a hard thing. Reconsider please if dependency should be injected